Core/Src/main.c

```diff
@@ -87,14 +89,38 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
+  MX_SPI1_Init();       // SPI is initialized here
+  MX_TIM1_Init();
   /* USER CODE BEGIN 2 */
 
+  uint8_t inputData[3] = {0x01, 0x80, 0x00};
+  uint8_t outputData[3] = {0};
+  uint16_t dataSize = 3;
+  uint32_t timeout = 10; // ms
+
+
   /* USER CODE END 2 */
 
   /* Infinite loop */
   /* USER CODE BEGIN WHILE */
+  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_SET); // Start with CS pin set or it may not work apparently
+  HAL_Delay(10);
   while (1)
   {
+	  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_RESET);  // CS pin low
+	  HAL_SPI_TransmitReceive(&hspi1, inputData, outputData, dataSize, timeout); // Ask for analog data as 3 bytes
+
+	  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_SET); // CS pin high before next SPI read
+
+	  uint16_t adcData = outputData[2] | ((outputData[1] & 0x03) << 8);  // Filter out the ADC data (only first 10 bits)
+
+	  // counts should range from 5% to 10% of the total period or from 3200 to 6400
+	  uint16_t pwmCounts = (adcData*25)/8 + 3200;  // convert the 0-1023 range ADC data to 3200-6396
```

Comment:
I have verified that this math works but I'm confused on where you got the *25 and /8 from. In any case, the 25, 8 and 3200 are considered magic numbers. They should be constants or #define with names so it is more clear for a reviewer like me to understand the math.